src/Adapter/FilesystemItems.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentApiBundle\Adapter;
+
+use CallbackFilterIterator;
+use EmptyIterator;
+use FilesystemIterator;
+use FluentDOM\Utility\Iterators\MapIterator;
+use InvalidArgumentException;
+use Iterator;
+use IteratorAggregate;
+use Libero\ContentApiBundle\Exception\ItemNotFound;
+use Libero\ContentApiBundle\Exception\UnexpectedVersionNumber;
+use Libero\ContentApiBundle\Exception\VersionNotFound;
+use Libero\ContentApiBundle\Model\ItemId;
+use Libero\ContentApiBundle\Model\ItemListPage;
+use Libero\ContentApiBundle\Model\Items;
+use Libero\ContentApiBundle\Model\ItemVersion;
+use Libero\ContentApiBundle\Model\ItemVersionNumber;
+use LimitIterator;
+use RuntimeException;
+use SplFileInfo;
+use Symfony\Component\Filesystem\Filesystem;
+use Traversable;
+use function array_pop;
+use function array_values;
+use function count;
+use function fopen;
+use function is_dir;
+use function iterator_count;
+use function iterator_to_array;
+use function md5_file;
+use function rsort;
+
+final class FilesystemItems implements IteratorAggregate, Items
+{
+    private $filesystem;
+    private $path;
+
+    public function __construct(string $path, Filesystem $filesystem)
+    {
+        $this->path = $path;
+        $this->filesystem = $filesystem;
+    }
+
+    public function add(ItemVersion $item) : void
+    {
+        $id = $item->getId();
+        $version = $item->getVersion();
+
+        try {
+            $next = $this->get($id)->getVersion()->next();
```

Comment:
could be broken up so that it's clear that `ItemNotFound` comes from `get()` rather than the other methods?